CaT/src/TableRelations/Graphs/Graph.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+namespace CaT\TableRelations\Graphs;
+/**
+ * Instead on relying on expensive graph search algorithms, 
+ * we will cache graph information in path-objcets as we build it.
+ * This means we will create for any node a set of paths starting
+ * from it, and update them successive as we add edges.
+ */
+class Graph implements AbstractGraph {
+
+	protected $nodes = array();
+	protected $edges = array();
+	public $connections = array();
+	protected $node_subgraph = array();
+	protected $paths = array();
+
+	/**
+	 * @inheritdoc
+	 */
+	public function addGraph(AbstractGraph $graph) {
```

Comment:
Die Methode können wir entfernen, oder? In Abstract Graph ist das nicht mehr enthalten.